sdk/core/core-amqp/test/config.spec.ts

```diff
@@ -235,6 +243,52 @@ describe("ConnectionConfig", function () {
         should.equal(config.entityPath, "entityPath", `EntityPath is not a string`);
       });
     });
+
+    describe("Development Emulator Validation", function () {
+      it("Accepts localhost", async function () {
+        const connectionString =
+          "Endpoint=localhost:6765;SharedAccessKeyName=<< REDACTED >>;SharedAccessKey=<< REDACTED >>;UseDevelopmentEmulator=true";
+        const config: ConnectionConfig = {
+          connectionString,
+          endpoint: "localhost:6765/",
+          host: "localhost:6765/",
+          sharedAccessKeyName: "sakName",
+          sharedAccessKey: "abcd",
+          useDevelopmentEmulator: true,
+        };
+        ConnectionConfig.validate(config);
+      });
+
+      it("Accepts 127.0.0.1", async function () {
```

Comment:
May be good to add test cases for IP v6 in there as well, as your regex handles them in its current form:
- `::1`
- `0:0:0:0:0:0:0:1`

Reply:
Good point, addressed in https://github.com/Azure/azure-sdk-for-js/pull/29195/commits/55799d8b3c5648dd001128e293e8b2f94fc7b013.